apps/core/.webpack/config.ts

```diff
@@ -0,0 +1,179 @@
+import { join, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { createRequire } from 'node:module';
+
+import TerserPlugin from 'terser-webpack-plugin';
+import webpack from 'webpack';
+
+import { productionCacheGroups } from './cache-group.js';
+
+const IN_CI = !!process.env.CI;
+
+export const rootPath = fileURLToPath(new URL('..', import.meta.url));
+
+const require = createRequire(rootPath);
+
+const packages = [
+  '@affine/component',
+  '@affine/i18n',
+  '@affine/debug',
+  '@affine/env',
+  '@affine/templates',
+  '@affine/workspace',
+  '@affine/jotai',
+  '@affine/copilot',
+  '@toeverything/hooks',
+  '@toeverything/y-indexeddb',
+];
+
+const isProduction = () => process.env.NODE_ENV === 'production';
+
+const OptimizeOptionOptions: () => webpack.Configuration['optimization'] =
+  () => ({
+    minimize: isProduction(),
+    minimizer: [
+      new TerserPlugin({
+        parallel: true,
+        extractComments: true,
+        terserOptions: {
+          parse: {
+            ecma: 2019,
+          },
+          compress: {
+            comparisons: false,
+          },
+          output: {
+            comments: false,
+            // https://github.com/facebookincubator/create-react-app/issues/2488
+            ascii_only: true,
+          },
+        },
+      }),
+    ],
+    removeEmptyChunks: true,
+    providedExports: true,
+    usedExports: true,
+    sideEffects: true,
+    removeAvailableModules: true,
+    runtimeChunk: {
+      name: 'runtime',
+    },
+    splitChunks: {
+      chunks: 'all',
+      minSize: 1,
+      minChunks: 1,
+      maxInitialRequests: Number.MAX_SAFE_INTEGER,
+      maxAsyncRequests: Number.MAX_SAFE_INTEGER,
+      cacheGroups: isProduction()
+        ? productionCacheGroups
+        : {
+            default: false,
+            vendors: false,
+          },
+    },
+  });
+
+export const createConfiguration: () => webpack.Configuration = () => {
+  let publicPath =
+    process.env.PUBLIC_PATH ?? (isProduction() ? '' : 'http://localhost:8080');
+  publicPath = publicPath.endsWith('/') ? publicPath : `${publicPath}/`;
+
+  return {
+    context: rootPath,
+    output: {
+      path: join(rootPath, 'dist'),
+      publicPath,
+    },
+
+    mode: isProduction() ? 'production' : 'development',
+
+    devtool: isProduction()
+      ? 'hidden-nosources-source-map'
```

Comment:
Should it be configurable? like debug source codes in electron app

Reply:
Working on it